File: 단계별로_풀어보기/6. 심화 1단계/6_2941.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

S = input().strip()
lenS = len(S)

croat2Arr = [
  'c=',
  'c-',
  'd-',
  'lj',
  'nj',
  's=',
  'z='
]

result = 0
idx = 0

while idx < lenS:
  if idx + 1 < lenS and S[idx:idx+2] in croat2Arr:
    idx += 1
  if idx + 2 < lenS and S[idx:idx+3] == 'dz=':
    idx += 2
  idx += 1
  result += 1

output(f"{result}")

File: 1. 단계별로_풀어보기/1~9/6. 심화 1단계/6_2941.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0, 'utf-8').toString().trim().split('\n')

// 입력 파싱
S = inputArr[0].trim().split('')

const lenS = S.length

const croat2Arr = ['c=', 'c-', 'd-', 'lj', 'nj', 's=', 'z=']

let result = 0

for (let i = 0; i < lenS; i++) {
  if (i + 1 < lenS) {
    const word2 = [S[i], S[i + 1]].join('')
    if (croat2Arr.includes(word2)) {
      i++
    } else if (i + 2 < lenS && [S[i], S[i + 1], S[i + 2]].join('') === 'dz=') {
      i += 2
    }
  }
  result += 1
}

process.stdout.write(`${result}`)
